### PR TITLE
ci: upload eslint annotations

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

make Logto great again

## Testing

see PR

### CODE REVIEW

File: merge-eslint-reports.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const fs = require('fs');
+
+const directories = fs.readdirSync('./packages');
+const reports = directories.map((dir) => fs.readFileSync(`./packages/${dir}/report.json`, { encoding: 'utf-8' }));
```
Comment (demonzoo) [inline merge-eslint-reports.js:4]: nit: we can use async read file method and wrapped in a Promise.all to await all files loaded

Comment (gao-sun) [reply merge-eslint-reports.js:4]: the outmost code block cannot use async

File: merge-eslint-reports.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+const merged = [];
+
+for (const report of reports) {
+  merged.push(...JSON.parse(report));
+}
```
Comment (demonzoo) [inline merge-eslint-reports.js:9]: nit: use reduce to avoid array.push

File: packages/console/package.json [hunk 3]
```diff
@@ -14,6 +14,7 @@
     "check": "tsc --noEmit",
     "build": "pnpm check && rm -rf dist && pnpm copyfiles && parcel build src/index.html --no-autoinstall --public-url /console",
     "lint": "eslint --ext .ts --ext .tsx src",
+    "lint:report": "pnpm lint -- --format json --output-file report.json",
```
Comment (demonzoo) [inline packages/console/package.json:17]: extra `--`?

File: merge-eslint-reports.js (hunk 1 above)
Comment (demonzoo) [reply merge-eslint-reports.js:4]: Ah, yes... you're right

File: merge-eslint-reports.js (hunk 2 above)
Comment (gao-sun) [reply merge-eslint-reports.js:9]: reduce is not clear in this case

Comment (gao-sun) [reply merge-eslint-reports.js:9]: we allowed reduce in our lint config

Comment (demonzoo) [reply merge-eslint-reports.js:9]: Yeah, I'm just nitpicking and trying to prevent you from merging this PR. :-)

Comment (gao-sun) [reply merge-eslint-reports.js:9]: hahaha

File: .github/workflows/upload-annotations.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        run: pnpm prepack
+
+      - name: Lint with Report
+        run: pnpm -- lerna run --parallel lint:report && node merge-eslint-reports.js
+
+      - name: Annotate Code Linting Results
+        uses: ataylorme/eslint-annotate-action@1.2.0
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          only-pr-files: false
```
Comment (simeng-li) [inline .github/workflows/upload-annotations.yml:29]: will it be more user friendly if set to only-pr-files true

Comment (gao-sun) [reply .github/workflows/upload-annotations.yml:29]: - one of the purposes of this PR is to catch legacy warnings
- actually this param doesn't work for now... will remove

Comment (gao-sun) [reply .github/workflows/upload-annotations.yml]: updated

**Review by simeng-li — APPROVED** [review]
LGTM

**Pull request merged**